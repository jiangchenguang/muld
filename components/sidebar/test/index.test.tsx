import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Sidebar from '..';
import SidebarItem from '../../sidebar-item';

describe('sidebar & sidebar-item', () => {
    afterEach(cleanup);
    it('click event & change event', () => {
        const onClick = jest.fn();
        const onChange = jest.fn();

        const { container } = render(
            <Sidebar onChange={onChange}>
                <SidebarItem>Text</SidebarItem>
                <SidebarItem onClick={onClick}>Text</SidebarItem>
            </Sidebar>,
        );

        const node = container.querySelectorAll('.mul-sidebar-item')[1];
        fireEvent.click(node);
        expect(onClick).toHaveBeenCalledWith(1);
        expect(onChange).toHaveBeenCalledWith(1);
    });

    it('click event & change event', () => {
        const onClick = jest.fn();
        const onChange = jest.fn();

        const { container } = render(
            <Sidebar onChange={onChange}>
                <SidebarItem>Text</SidebarItem>
                <SidebarItem onClick={onClick}>Text</SidebarItem>
            </Sidebar>,
        );

        const node = container.querySelectorAll('.mul-sidebar-item')[1];
        fireEvent.click(node);
        expect(onClick).toHaveBeenCalledWith(1);
        expect(onChange).toHaveBeenCalledWith(1);
    });

    it('disabled prop', () => {
        let activeKey = 0;
        const { container, asFragment } = render(
            <Sidebar
                activeKey={activeKey}
                onChange={(i: number) => {
                    activeKey = i;
                }}
            >
                <SidebarItem>Text</SidebarItem>
                <SidebarItem disabled>Text</SidebarItem>
            </Sidebar>,
        );

        const node = container.querySelectorAll('.mul-sidebar-item')[1];
        fireEvent.click(node);
        expect(activeKey).toEqual(0);

        expect(asFragment()).toMatchSnapshot();
    });
});
