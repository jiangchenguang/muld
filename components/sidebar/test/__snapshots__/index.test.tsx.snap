// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar & sidebar-item disabled prop 1`] = `
<DocumentFragment>
  .c0.mul-sidebar {
  width: 80px;
  overflow-y: auto;
  -webkit-overflow-scrolling: touch;
}

.c1.mul-sidebar-item {
  position: relative;
  display: block;
  box-sizing: border-box;
  padding: 20px 12px;
  overflow: hidden;
  color: #323233;
  font-size: 14px;
  line-height: 20px;
  word-wrap: break-word;
  background-color: #f7f8fa;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1.mul-sidebar-item:active {
  background-color: #f2f3f5;
}

.c1.mul-sidebar-item:not(:last-child)::after {
  border-bottom-width: 1px;
}

.c1.mul-sidebar-item .mul-sidebar-item__text {
  position: relative;
  display: inline-block;
}

.c1.mul-sidebar-item--select {
  color: #323233;
  font-weight: 500;
}

.c1.mul-sidebar-item--select,
.c1.mul-sidebar-item--select:active {
  background-color: #fff;
}

.c1.mul-sidebar-item--select::before {
  position: absolute;
  top: 50%;
  left: 0;
  width: 4px;
  height: 16px;
  background-color: #fa220a;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  content: '';
}

.c1.mul-sidebar-item--disabled {
  color: #c8c9cc;
  cursor: not-allowed;
}

.c1.mul-sidebar-item--disabled:active {
  background-color: #f7f8fa;
}

<div
    class="c0 mul-sidebar"
  >
    <a
      class="c1 mul-sidebar-item mul-sidebar-item--select"
    >
      <div
        class="mul-sidebar-item__text"
      />
    </a>
    <a
      class="c1 mul-sidebar-item mul-sidebar-item--disabled"
    >
      <div
        class="mul-sidebar-item__text"
      />
    </a>
  </div>
</DocumentFragment>
`;
